Sources/Commands/SwiftRunCommand.swift

```diff
@@ -331,36 +335,30 @@ public struct SwiftRunCommand: AsyncSwiftCommand {
         #if !os(Windows)
         // Dispatch will disable almost all asynchronous signals on its worker threads, and this is called from `async`
         // context. To correctly `exec` a freshly built binary, we will need to:
-        // 1. reset the signal masks
+        // 1. Reset the signal masks
         for i in 1..<NSIG {
             signal(i, SIG_DFL)
         }
         var sig_set_all = sigset_t()
         sigfillset(&sig_set_all)
         sigprocmask(SIG_UNBLOCK, &sig_set_all, nil)
 
-        #if os(FreeBSD) || os(OpenBSD)
-        #if os(FreeBSD)
-        pthread_suspend_all_np()
-        #endif
-        closefrom(3)
-        #else
-        #if os(Android)
-        let number_fds = Int32(sysconf(_SC_OPEN_MAX))
-        #else
-        let number_fds = getdtablesize()
-        #endif /* os(Android) */
-        
-        // 2. close all file descriptors.
-        for i in 3..<number_fds {
-            close(i)
-        }
-        #endif /* os(FreeBSD) || os(OpenBSD) */
+        // 2. Close all file descriptors above stdio.
+        closeFdsAbove(3)
         #endif
 
         try TSCBasic.exec(path: path, args: args)
```

Comment:
I realise that doesn't originate from your new code but I think it's super dangerous to run Swift between `fork()` and `execve()`. We're essentially in a broken process, so this might just hang. For example this situation could happen:

- program runs happily
- Thread N acquires some lock (e.g. an allocator lock or a Swift runtime lock)
- `fork()` gets called on thread M
- new process starts, continuing to run thread M exclusively
- the Swift code here allocates or calls into the Swift runtime
- can't acquire the lock because it's marked as already in use and will never be freed because the acquisition happened in the old process.

Reply:
Yea, fork cludgery in a runtime based language is always tricky. Fwiw we do allocate in opendir, although we could swap to getdents64 and whatever the equivalent is on macOS.

Reply:
Y'know, I just took a look outside of this code, and I don't even see anywhere that fork takes place. spm might just be doing execve directly with no fork?

Reply:
Iiinteresting, yes, for `swift run` that's actually possible. Of course that theoretically opens up a whole can of works of course because we're multi-threaded. So all the signal masking/resetting stuff is technically racy unless we do the equivalent of `pthread_suspend_all_np()`.

But of course, none of this is the issue of this PR. Maybe leave a bunch of comments about what we're aiming to do and why the suspension of the other threads is now necessary (but harder to do in at least Linux which doesn't have `pthread_suspend_all_np` from what I know)...

Reply:
Added a lengthy comment on the original reason this was added (dug through the history, the change was recent also)